modules/network/vpc/main.tf

```diff
@@ -15,8 +15,9 @@
 */
 
 locals {
-  network_name    = var.network_name == null ? "${var.deployment_name}-net" : var.network_name
-  subnetwork_name = var.subnetwork_name == null ? "${var.deployment_name}-primary-subnet" : var.subnetwork_name
+  autoname        = replace(var.deployment_name, "_", "-")
```

Comment:
Shouldn't deployment name never contain underscores?  If so, why don't we just throw an error instead during validation?

Reply:
> Shouldn't deployment name never contain underscores?

No, it just needs to be a valid GCP label:
https://cloud.google.com/resource-manager/docs/creating-managing-labels#requirements